File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F15.P1.go
package main

import "fmt"

// BEGIN PROBE F15.P1
func head[T any](xs []T) T { return xs[0] }

func combine() string {
	a := head([]int32{4, 5, 6})
	b := head([]string{"p", "q"})
	return fmt.Sprint(a) + b
}

// END PROBE F15.P1

func main() {
	fmt.Println(combine())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/python/F15.P1.py
# BEGIN PROBE F15.P1
def head[T](values: list[T]) -> T:
    return values[0]


a = head([4, 5, 6])
b = head(["p", "q"])
result = str(a) + b
# END PROBE F15.P1

print(result)
